sql/timeseries/h3.sql

```diff
@@ -0,0 +1,32 @@
+#standardSQL
+# The amount of requests either using HTTP/3 or able to use it.
+#
+# We measure "ability to use" as well as "actual use", as HTTP Archive is a
+# cold crawl and so less likely to use HTTP/3 which requires prior visits.
+# 
+# For "able to use" we look at the alt-svc response header.
+#
+# We also only measure official HTTP/3 (ALPN h3, h3-29) and not gQUIC or other
+# prior versions. h3-29 is the final draft version and will be switched to h3
+# when HTTP/3 is approved so we include that as it is HTTP/3 in all but name.
+#
+SELECT
+  SUBSTR(_TABLE_SUFFIX, 0, 10) AS date,
+  UNIX_DATE(CAST(REPLACE(SUBSTR(_TABLE_SUFFIX, 0, 10), '_', '-') AS DATE)) * 1000 * 60 * 60 * 24 AS timestamp,
+  IF(ENDS_WITH(_TABLE_SUFFIX, 'desktop'), 'desktop', 'mobile') AS client,
+  ROUND(SUM(IF(respHttpVersion = 'HTTP/3' OR respHttpVersion = 'h3' OR respHttpVersion = 'h3-29'
+                OR reqHttpVersion = 'HTTP/3' OR reqHttpVersion = 'h3' OR reqHttpVersion = 'h3-29'
+                OR REGEXP_EXTRACT(REGEXP_EXTRACT(respOtherHeaders, r'alt-svc = (.*)'), r'(.*?)(?:, [^ ]* = .*)?$') LIKE '%h3=%'
+                OR REGEXP_EXTRACT(REGEXP_EXTRACT(respOtherHeaders, r'alt-svc = (.*)'), r'(.*?)(?:, [^ ]* = .*)?$') LIKE '%h3-29=%'
```

Comment:
Can these be simplified at all, for example a single REGEXP_CONTAINS function call? Or does combining all this regex into one defeat the purpose of simplifying? :p

Reply:
This is why we need JSON header fields!

Just concerned that `h3` is quite a small search string so wanted to ensure it definitely was only in the `alt-svc` header field and not any other field (e.g. . Unfortunately the `alt-svc` header, also contains commas, spaces and equals, and which makes this complex. I couldn't figure out a nice way to do this other than one REGEX_EXTRACT to get the `alt-svc` field onwards, and then another regex to strip out the following fields.

`.*alt-svc = (.*?)(?:, [^ ]* = .*)?$` seems to work on https://regex101.com/r/a9b0r9/1 but not in BigQuery.

Got any better way of doing this?

Reply:
I'm sure it can be done but probably not worth the effort. Happy to resolve this and merge.